test/hmactest.c

```diff
@@ -284,7 +284,7 @@ static char *pt(unsigned char *md, unsigned int len)
     if (md == NULL)
         return NULL;
     for (i = 0; i < len; i++)
-        sprintf(&(buf[i * 2]), "%02x", md[i]);
+        BIO_snprintf(&(buf[i * 2]), sizeof(buf), "%02x", md[i]);
```

Comment:
`sizeof(buf)` is wrong, since we are writing to `buf + i * 2`. Either `sizeof(buf) - i * 2` or `2` would be better.

Note: we should also really check `len` here (but that's probably out of scope for this PR)

Reply:
See above.